langchain/vectorstores/mongodb_atlas.py

```diff
@@ -0,0 +1,242 @@
+            from langchain.embeddings.openai import OpenAIEmbeddings
+            from pymongo import MongoClient
+
+            mongo_client = MongoClient("<YOUR-CONNECTION-STRING>")
+            namespace = "<db_name>.<collection_name>"
+            embeddings = OpenAIEmbeddings()
+            vectorstore = MongoDBAtlasVectorSearch(mongo_client, namespace, embeddings)
+    """
+
+    def __init__(
+        self,
+        client: MongoClient,
+        namespace: str,
+        embedding: Embeddings,
+        *,
+        index_name: str = "default",
+        text_key: str = "text",
+        embedding_key: str = "embedding,",
+    ):
+        """
+        Args:
+            client: MongoDB client.
+            namespace: MongoDB namespace to add the texts to.
+            embedding: Text embedding model to use.
+            text_key: MongoDB field that will contain the text for each
+                document.
+            embedding_key: MongoDB field that will contain the embedding for
+                each document.
+        """
+        self._client = client
+        db_name, collection_name = namespace.split(".")
+        self._collection = client[db_name][collection_name]
+        self._embedding = embedding
+        self._index_name = index_name
+        self._text_key = text_key
+        self._embedding_key = embedding_key
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        connection_string: str,
+        namespace: str,
+        embedding: Embeddings,
+        **kwargs: Any,
+    ) -> MongoDBAtlasVectorSearch:
+        try:
+            from pymongo import MongoClient
+        except ImportError:
+            raise ImportError(
+                "Could not import pymongo, please install it with "
+                "`pip install pymongo`."
+            )
+        client = MongoClient(connection_string)
+        return cls(client, namespace, embedding, **kwargs)
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Run more texts through the embeddings and add to the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+
+        Returns:
+            List of ids from adding the texts into the vectorstore.
+        """
+        # Embed and create the documents
+        docs = []
+        for i, text in enumerate(texts):
+            metadata = metadatas[i] if metadatas else {}
+            embedding = self._embedding.embed_documents([text])[0]
+            doc = {self._text_key: text, self._embedding_key: embedding, **metadata}
+            docs.append(doc)
+
+        # insert in MongoDB Atlas
+        insert_result = self._collection.insert_many(docs)
+        return insert_result.inserted_ids
+
+    def similarity_search_with_score(
+        self,
+        query: str,
+        *,
```

Comment:
Just wanted to provide the following information: in MongoDB, a namespace (database + collection - like a table in SQL) can have several indexes. By removing the parameters allowing the user to change the default namespace and index, it means we are making a one to one mapping between a tuple of (namespace, index) and a vectorstore instance (MongoDBAtlasVectorSearch). So if the user wants to perform a similarity search in a different namespace and index, they will have to create a new instance.

Reply:
i think having one vector store instance map to one table sounds ok to me, unless there's a very high cost to creating new instances?

Reply:
Yes you're right. That's more clear